### PR TITLE
Add common protection api support & small fixes

### LINKED ISSUE #22: Bombarder not respecting FLAN claims
1.8.1 Blast / 1.17.1 / Fabric (latest API)

Two things are happening.

1) Bombarder is being destructive but for some reason only in FLAN claims.
2) It's not respecting FLAN claims and destroying claimed blocks

### LINKED ISSUE #34: FTB Chunks + BLAST
You can use the pearl bomb in ftb claims, I have not checked the other bombs. I would like to report this to them directly but they seem to have no way to

Playing: all of fabric 5 1.5.1

### LINKED ISSUE #44: The pipe bomb entity does not have a translation
Running `/summon blast:pipe_bomb` reveals that the pipe bomb entity is not translated, using the `entity.blast.pipe_bomb` fallback string instead.

### PR DESCRIPTION
**Pull request by EvieTheOwl**
Added support for Patbox's common protection API (https://github.com/Patbox/common-protection-api) which prevents explosions bypassing claims for land claiming mods like Cadmus (https://modrinth.com/mod/cadmus)

Also fixes #44 and an issues where you could get floating fire blocks on the client

### CODE REVIEW

File: src/main/java/ladysnake/blast/common/block/BonesburrierBlock.java [hunk 1]
```diff
@@ -43,9 +43,12 @@ public static void prime(World world, BlockPos pos) {
     private static void prime(World world, BlockPos pos, LivingEntity igniter) {
         if (!world.isClient && world.getBlockState(pos).getBlock() instanceof BonesburrierBlock) {
             BonesburrierEntity entity = BlastEntities.BONESBURRIER.create(world);
-            entity.setPos(pos.getX() + 0.5, pos.getY(), pos.getZ() + 0.5);
-            world.spawnEntity(entity);
-            world.playSound(null, entity.getX(), entity.getY(), entity.getZ(), SoundEvents.ENTITY_TNT_PRIMED, SoundCategory.BLOCKS, 1.0F, 1.0F);
+            if (entity != null) {
```
Comment (Pyrofab) [inline src/main/java/ladysnake/blast/common/block/BonesburrierBlock.java]: Should probably error here if it is in fact null

**Review by Pyrofab — COMMENTED** [review]
Will review this more in depth later but at first glance it's good work again - just makes me think we should really refactor the block classes at some point to reduce the code duplication

**Comment by EvieTheOwl** [comment]
> Will review this more in depth later but at first glance it's good work again - just makes me think we should really refactor the block classes at some point to reduce the code duplication

True. That's gonna take a bit longer I think. Will of course be worth switching to variables that mean something as well

**Comment by EvieTheOwl** [comment]
Video of it working: https://youtu.be/6LrdmD8crsg

**Comment by sleepy-evelyn** [comment]
Now fixes #44 & a bug where fire could be seen floating on the client after exploding gunpowder

**Comment by EvieTheOwl** [comment]
Added documentation to the readme and sent a PR for the wiki: https://github.com/Ladysnake/Ladysnake.github.io/pull/77.
Unless any issues or changes are required should be finished with this PR now

File: src/main/java/ladysnake/blast/common/Blast.java [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
 import ladysnake.blast.common.init.BlastItems;
 import ladysnake.blast.common.init.BlastSoundEvents;
 import net.fabricmc.api.ModInitializer;
+import net.minecraft.entity.damage.DamageSource;
 import net.minecraft.entity.data.TrackedDataHandler;
 import net.minecraft.entity.data.TrackedDataHandlerRegistry;
 import net.minecraft.network.PacketByteBuf;
+import net.minecraft.server.network.ServerPlayerEntity;
 import net.minecraft.util.math.Direction;
+import org.jetbrains.annotations.Nullable;
```
Comment (Pyrofab) [inline src/main/java/ladysnake/blast/common/Blast.java]: Are these imports used anywhere ? I do not see any change elsewhere in the file

File: src/main/java/ladysnake/blast/common/entity/ShrapnelBlockEntity.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        return explosionThrownBlockEntity;
+    }
+
+    @Nullable
+    public PlayerEntity getOwner() { return owner; }
+
+    @Override
+    protected void writeCustomDataToNbt(NbtCompound nbt) {
+        super.writeCustomDataToNbt(nbt);
+        if (!getWorld().isClient && owner != null)
```
Comment (Pyrofab) [inline src/main/java/ladysnake/blast/common/entity/ShrapnelBlockEntity.java]: Could you add braces for those 2-lines statements ? It's a little thing but I think it improves homogeneity and prevents the stupidest kind of bugs.

File: src/main/java/ladysnake/blast/common/block/StripminerBlock.java [hunk 4]
```diff
@@ -44,10 +44,12 @@ public static void primeStripminer(World world, BlockPos pos) {
     private static void primeStripminer(World world, BlockPos pos, LivingEntity igniter) {
         if (!world.isClient && world.getBlockState(pos).getBlock() instanceof StripminerBlock) {
             StripminerEntity entity = ((StripminerBlock) world.getBlockState(pos).getBlock()).type.create(world);
+            entity.setOwner(igniter);
             entity.setFacing(world.getBlockState(pos).get(FACING));
             entity.setPos(pos.getX() + 0.5, pos.getY(), pos.getZ() + 0.5);
             world.spawnEntity(entity);
             world.playSound(null, entity.getX(), entity.getY(), entity.getZ(), SoundEvents.ENTITY_TNT_PRIMED, SoundCategory.BLOCKS, 1.0F, 1.0F);
+
```
Comment (Pyrofab) [inline src/main/java/ladysnake/blast/common/block/StripminerBlock.java]: Why the extra space ?

**Review by Pyrofab — COMMENTED** [review]
Not easy to review with all the existing decompiled code, but looks good to me. Just had to annoy you with some code style things, which by the way reminded me to add an editorconfig to the project.

**Review by Pyrofab — APPROVED** [review]
Very cool thanks

**Pull request merged**